src/ui.rs

```diff
@@ -202,8 +279,26 @@ impl Ui {
     }
 
     pub fn flush(&mut self) -> io::Result<()> {
-        match &mut self.output_pair {
-            UiOutputPair::Terminal { stdout, .. } => stdout.flush(),
+        match &mut self.output {
+            UiOutput::Terminal { stdout, .. } => stdout.flush(),
+            UiOutput::Paged { child_stdin, .. } => child_stdin.flush(),
+        }
+    }
+
+    pub fn finalize_writes(&mut self) {
+        if let UiOutput::Paged {
+            mut child,
+            child_stdin,
+        } = mem::replace(&mut self.output, UiOutput::new_terminal())
+        {
+            drop(child_stdin);
+            child
+                .wait()
+                // It's possible (though unlikely) that this write fails, but
+                // this function gets called so late that there's not much we
+                // can do about it.
+                .map_err(|e| self.write_error(&format!("Failed to wait on pager '{}'", e)))
+                .ok();
```

Comment:
This could be written more cleanly as `if let Err(e) = child.wait() { ... }`. Prefer to only use `map`-like functions when transforming contained data into a new form that will be processed further downstream.